std/functional.d

```diff
@@ -899,6 +903,89 @@ template partial(alias fun, alias arg)
     assert(result2 == 1.5f);
 }
 
+/**
+Takes a function of (potentially) many arguments, and returns a function taking
+one argument and returns a callable taking the rest.  f(x, y) == curry(f)(x)(y)
+
+Params:
+    F = a function taking at least one argument
+Returns:
+    A single parameter callable object
+*/
+template curry(alias F)
+{
+	//adapted from the implementation from Artur Skawina here:
+	//https://forum.dlang.org/post/mailman.1626.1340110492.24740.digitalmars-d@puremagic.com
+	auto curry()
+	{
+		struct CurryImpl {
+			import std.traits;
+			Parameters!F args;
+			template opCall1(size_t N=0)
+			{
+				auto ref opCall1(Parameters!F[N] a)
+				{
+					auto p = (cast(Unqual!(typeof(a))*)&args[N]);
+					*p = a;
+					static if (N==args.length-1)
+					{
+						return F(args);
+					}
+					else
+						return &opCall1!(N+1);
+				}
+			}
+			alias opCall1!() opCall;
```

Comment:
This alias style is deprecated. Can we use the newer one with assignment?

Reply:
Fixed